docs/clouds/qemu.md

```diff
@@ -0,0 +1,104 @@
+# QEMU
+
+The following page documents the QEMU cloud integration in pycloudlib.
+
+## Support
+
+Pycloudlib currently supports launching x86-64 Ubuntu images using
+`qemu-system-x86_64` on an x86-64 host.
+
+## Requirements
+
+In addition to requirements specified in `setup.py`, on Ubuntu systems,
+the QEMU cloud requires the following packages:
+
+* qemu-system-x86
+* qemu-utils  # for qemu-img
+* genisoimage  # to create the cloud-init NoCloud seed image
+
+## Configuration and Local Storage
+
+QEMU instances require local directories to store images and instances.
+These directories can be configured in the `pycloudlib.toml` file:
+
+`image_dir` - Directory to look for images when launching instances.
+  Also the directory to download images to where supported.
+
+`working_dir` - Directory to store temporary instance data. This includes
+  instance disks, snapshots, and other metadata.
+
+`qemu_binary` - Path to the `qemu-system-x86_64` binary.
+
+## Obtaining Images
+
+While any x64-64 should be usable, Ubuntu images can be downloaded from the
+Ubuntu cloud images website using the cloud `daily_image` and `released_image`
+calls.
+
+For example:
+
+```python
+cloud = Qemu(...)
+cloud.released_image("jammy")
+```
+
+will return the path to the latest jammy image that has already been
+downloaded by Pycloudlib. If no image exists, it will be downloaded
+into the image_dir.
+
+```python
+cloud = Qemu(...)
+cloud.daily_image("jammy")
+```
+
+will return the path to the latest jammy image that has been released.
+If the image does not exist, it will be downloaded into the image_dir.
+
+## Launching Instances
+
+Launching instances with QEMU requires an `image_id`.
+
+The `image_id` can either be an absolute path image, or a path relative
+to the `image_dir`. It is most easily retrieved using the `daily_image` or
+`released_image` calls.
+
+The `instance_type` is in the form of `c<cpus>m<memory>`, where `<cpus>` is
+the number of CPUs to allocate to the instance and `<memory>` is the amount
+of memory to allocate to the instance in megabytes.
+
+If `kernel_cmdline` is specified, it will be passed to the kernel as the
+kernel command line. When specified, `kernel_path` must also be specified.
+The `daily_image` and `released_image` calls will automatically download
+the kernel for the image, so if these calls have been used to obtain the
+image_id, the `kernel_path` is not required.
+
+By default, pycloudlib will create a seed iso for the instance in order
+to let cloud-init configure the instance. If `no_seed_iso` is set to
+`True`, pycloudlib will not create a seed iso. This may be useful when
+using the `kernel_cmdline` behavior or when using an image that already
+has keys and/or passwords configured.
```

Comment:
I think we should only have separate code samples if the cloud provides additional API features on top of the core API. It doesn't here. I think it would be good to have an overarching example that can work on multiple clouds though.